src/filemap.rs

```diff
@@ -167,6 +167,15 @@ where
                 return Ok(has_diff);
             }
         }
+        WriteMode::Modified => {
+            let filename = filename_to_path();
```

Comment:
I think we can just do nothing here, or just do the `has_diff` part and then remove `output_modified` since I think the important bit of this work is returning the info from `get_modified_lines` rather than writing something out as in other write modes.

Reply:
Writing to `out` here is needed as that's how `get_modified_lines` gets its results (it's passed down via `format_input`), as there's no other way (as `format_input` exists at the moment) to pass the results back other than through this `T: Write`.

I did look initially at factoring the guts of `format_input` out in the hope of sharing that with `get_modified_lines` but it seemed hard to avoid ending up with two fairly long and very similar functions.  I can look again if you think that's worth it.